### DISCUSSION TITLE
Null/undefined values

### DISCUSSION THREAD

**Discussion by wobsoriano** [DamianOsipiuk/vue-query#184] [Q&A]
So I'm using another subscription library and do want to know if there's a way vue-query (or vue in general) can detect changes when set to null or an empty object?

```ts
const unwatch = watchAccount((data) => {
  queryClient.setQueryData(queryKey(), data?.address ? data : null)
})
```

The code above won't trigger updates because of weird vue reactivity. But do you have any suggestions for things like this? 😄

**Comment by DamianOsipiuk** [comment]
Hmm I think in this case you would like to remove the query from cache instead of setting it to null ([queryClient.removeQueries](https://react-query.tanstack.com/reference/QueryClient#queryclientremovequeries)).
Would that solve your usecase?
